From 321d93aaf32baa3338a44e9beb56594d728d46db Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Mon, 9 Sep 2024 16:16:49 +0200
Subject: [PATCH] Input: goodix - add delay after reset

Add a delay between probe startup and first i2c reading
to ensure the touchscreen is in a stable hardware state
(the reset & the pin interrupt set by pin control) .

Change-Id: I1fec90121042d9a8075ff89c1df71f6c94515ac4
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/403184
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/input/touchscreen/goodix.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/input/touchscreen/goodix.c b/drivers/input/touchscreen/goodix.c
index a94b8f5fac56..fd40c27e9cde 100644
--- a/drivers/input/touchscreen/goodix.c
+++ b/drivers/input/touchscreen/goodix.c
@@ -993,7 +993,6 @@ static int goodix_get_gpio_config(struct goodix_ts_data *ts)
 				     GOODIX_GPIO_RST_NAME);
 
 	ts->gpiod_rst = gpiod;
-
 	switch (ts->irq_pin_access_method) {
 	case IRQ_PIN_ACCESS_ACPI_GPIO:
 		/*
@@ -1371,6 +1370,9 @@ static int goodix_ts_probe(struct i2c_client *client)
 		error = goodix_reset(ts);
 		if (error)
 			return error;
+	} else {
+		/* Need to wait at least 50 ms */
+		msleep(50);
 	}
 
 	error = goodix_i2c_test(client);
-- 
2.39.5

